xbmc/pvr/windows/GUIWindowPVRGuide.cpp

```diff
@@ -46,14 +46,13 @@ CGUIWindowPVRGuide::CGUIWindowPVRGuide(bool bRadio) :
   m_refreshTimelineItemsThread(new CPVRRefreshTimelineItemsThread(this)),
   m_bRefreshTimelineItems(false)
 {
-  m_cachedTimeline = new CFileItemList;
+  m_cachedTimeline = std::make_shared<CFileItemList>();
```

Comment:
Actually, the most efficient way to do it is

    m_cachedTimeline.reset(new CFileItemList());

Since this even avoids the `shared_ptr` move constructor with a temporary. Less readable, but most efficient.

There is no problem with exception-safety. If the constructor throws, the allocation will be deleted, and the `shared_ptr` will be left unchanged. That applies to all 3 variants.

(According to http://en.cppreference.com/w/cpp/memory/shared_ptr/make_shared, `make_shared()` is C++03 or older.)